Sources/Apodini/Response/Status.swift

```diff
@@ -7,18 +7,33 @@
 //              
 
 import NIO
+import NIOHTTP1
 
 
 /// An `Status` expresses additional information that can be passed to a `Response`
-public enum Status: ResponseTransformable {
+public enum Status: ResponseTransformable, Equatable {
     /// The request was handled and the response contains the expected content
     case ok
     /// The request was handled and a new resource has been created
     case created
     /// The request was handled and the response does not contain any content
     case noContent
+    /// The request was accepted and an asynchronous action has been started
+    case accepted
+    
     /// The request was handled and the client should be redirected to another URL
     case redirect
+    /// The request was handled and the response has not been changed versus the client's cache
+    case notModified
+    
+    /// The request was not handled as the request was malformed
+    case badRequest
+    /// The request was not handled as the resource could not be found
+    case notFound
+    /// The request could not be processed as it is in conflict with the resource's state
+    case conflict
+    /// A conditional operation has failed, e.g. the client's state is outdated.
+    case preconditionFailed
```

Comment:
In Apodini, error cases and the corresponding status codes are handles using an `ApodiniError` and the `@Throws` property wrapper. I would suggest to utilize Apodini errors in the corresponding cases instead of extending the `Status` enum with failure cases. This would allow us to keep up that distinction between success and error states.